sphinx/application/data/features/feature-repository/src/main/java/chat/sphinx/feature_repository/SphinxRepository.kt

```diff
@@ -3906,31 +3906,34 @@ abstract class SphinxRepository(
         val token = memeServerTokenHandler.retrieveAuthenticationToken(memeServerHost)
             ?: throw RuntimeException("MemeServerAuthenticationToken retrieval failure")
 
-        networkQueryMemeServer.getPaymentTemplates(token, moshi = moshi).collect { loadResponse ->
-            @Exhaustive
-            when (loadResponse) {
-                is LoadResponse.Loading -> {}
-                is Response.Error -> {
-                    response = loadResponse
-                }
-                is Response.Success -> {
-                    var templates = ArrayList<PaymentTemplate>()
-
-                    for (ptDto in loadResponse.value) {
-                        templates.add(
-                            PaymentTemplate(
-                                ptDto.muid,
-                                ptDto.width,
-                                ptDto.height,
-                                token.value
-                            )
-                        )
+        try {
+            networkQueryMemeServer.getPaymentTemplates(token, moshi = moshi).collect { loadResponse ->
+                @Exhaustive
+                when (loadResponse) {
+                    is LoadResponse.Loading -> {
+                    }
+                    is Response.Error -> {
+                        response = loadResponse
```

Comment:
remove unnecessary try/catch block and add annotation for throwing RuntimeException on line 3907, or remove the exception in favor of doing a null check and returning an error.